client_src/js/lib/octoprint.js

```diff
@@ -13,7 +13,8 @@ export default class OctoPrintClient {
 
   static get(printer, item) {
     this.validatePrinter(printer);
-    const url = `${printer.printerURL}/${item}`;
+
+    const url = new URL(item, printer.printerURL).href;
```

Comment:
@NotExpectedYet let me know if this conflicts. If so Ill revert it from this PR.

Reply:
@davidzwa It will do as I removed all the get's out of octoprint-client.js. I also changed it's name and put it in services. 

The URL checking is done before axios is called as well so you don't need to repeat it. Was going to try and do the same for validate printer too.